packages/components/menu/src/menu.ts

```diff
@@ -412,20 +429,59 @@ export default defineComponent({
 
       const ulStyle = useMenuCssVar(props, 0)
 
-      const vMenu = h(
-        'ul',
-        {
-          key: String(props.collapse),
-          role: 'menubar',
-          ref: menu,
-          style: ulStyle.value,
-          class: {
-            [nsMenu.b()]: true,
-            [nsMenu.m(props.mode)]: true,
-            [nsMenu.m('collapse')]: props.collapse,
+      const constructIndexPath = (index: string) => {
+        const indexPath: string[] = []
+        const splits = index.split('-')
+
+        for (let i = 0; i < splits.length; i++) {
+          const nextIndexPath = splits.slice(0, i + 1).join('-')
+          indexPath.push(nextIndexPath)
+        }
+
+        return indexPath
+      }
```

Comment:
This way of construct the index path may not be the actual path. The User defined index may not be separated by `-`.

Reply:
Yep my bad, I don't realize that I can get `indexPath` from the `subMenus`. I clean up my stuff and update a new code. Thank you so much 👍